src/python/grpcio/grpc/experimental/aio/_call.py

```diff
@@ -58,22 +59,17 @@ class AioRpcError(grpc.RpcError):
     determined. Hence, its methods no longer needs to be coroutines.
     """
 
-    # TODO(https://github.com/grpc/grpc/issues/20144) Metadata
-    # type returned by `initial_metadata` and `trailing_metadata`
-    # and also taken in the constructor needs to be revisit and make
-    # it more specific.
-
     _code: grpc.StatusCode
     _details: Optional[str]
-    _initial_metadata: Optional[MetadataType]
-    _trailing_metadata: Optional[MetadataType]
+    _initial_metadata: Optional[Metadata]
+    _trailing_metadata: Optional[Metadata]
     _debug_error_string: Optional[str]
 
     def __init__(self,
                  code: grpc.StatusCode,
                  details: Optional[str] = None,
-                 initial_metadata: Optional[MetadataType] = None,
-                 trailing_metadata: Optional[MetadataType] = None,
+                 initial_metadata: Optional[Metadata] = None,
+                 trailing_metadata: Optional[Metadata] = None,
```

Comment:
Can be them set to `None`? If so the return type for the getter methods would need to be changed, for example [1], if not we should remove the `Optional` (the same for the class type definition)

https://github.com/grpc/grpc/pull/23045/files#diff-802047f7d320089cb1d9c7de77c9cae5R107

Reply:
Good point, I think we can remove the ``Optional[]`` because it's always a ``Metadata`` object, however, I wouldn't remove the default argument ``= None`` because that would imply changing the position of the parameters (they would have to go right after ``code``), and that could cause backward incompatibilities, and also confuse users.
The proposal is then to remove the ``Optional[..]`` annotation but keep the default argument to ``None``. WDYT?

Reply:
weird :), If we keep the `= None` I will keep the `Optional`, regarding backward compatibility the constuctor of `AioRpcError` is internal, so it menas that If Im not wrong we can turn it to a positional argument.

WDYT @lidizheng ?

Reply:
To me, `Optional` looks better in terms of conveying the variable is "optional".

Yes, this is an internal API, users won't be impacted. I don't have strong opinion here, either way looks fine to me. Since @rmariano is the author of the PR, maybe he should make the call?

Reply:
up to you @rmariano !

Reply:
I'll refactor this, to make these parameters no longer optional.